configuration/wireless/access-point-routed.md

```diff
@@ -0,0 +1,193 @@
+
+```
+sudo apt install dnsmasq
+```
+Finally, install `netfilter-persistent` and its plugin `iptables-persistent`. This utilty helps by saving firewall rules and restoring them when the Raspberry Pi boots:
+
+```
+sudo DEBIAN_FRONTEND=noninteractive apt install -y netfilter-persistent iptables-persistent
+```
+Software installation is complete. We will configure the software packages later on.
+
+<a name="routing"></a>
+## Set up the network router
+
+The Raspberry Pi will run and manage a standalone wireless network. It will also route between the wireless and Ethernet networks, providing internet access to wireless clients. If you prefer, you can choose to skip the routing by skipping the section "Enable routing and IP masquerading" below, and run the wireless network in complete isolation. 
+
+### Define the wireless interface IP configuration
+
+In this document, we assume IP network `10.10.0.0/24` is configured on an Ethernet LAN, and the Raspberry Pi will manage IP network `192.168.4.0/24` for wireless clients.
+**Note:** Please select another IP network for wireless, e.g. `192.168.10.0/24`, if IP network `192.168.4.0/24` is already in use by your Ethernet LAN.
+
+The Raspberry Pi runs a DHCP server for the wireless network; this requires static IP configuration for the wireless interface (`wlan0`) in the Raspberry Pi. 
+The Raspberry Pi also acts as the router on the wireless network, and as is customary, we will give it the first IP address in the network: `192.168.4.1`.
+
+To configure the static IP address, edit the configuration file for `dhcpcd` with:
+
+```
+sudo nano /etc/dhcpcd.conf
+```
+
+Go to the end of the file and add the following:
+
+```
+interface wlan0
+    static ip_address=192.168.4.1/24
+    nohook wpa_supplicant
+```
+
+### Enable routing and IP masquerading
+
+This section configures the Raspberry Pi to let wireless clients access computers on the main (Ethernet) network, and from there the internet.
+**NOTE:** If you wish to block wireless clients from accessing the Ethernet network and the internet, skip this section. 
+
+To enable routing, i.e. to allow traffic to flow from one network to the other in the Raspberry Pi, create a file using the following command, with the contents below:
+```
+sudo nano /etc/sysctl.d/routed-ap.conf
+```
+File contents:
+```
+# https://www.raspberrypi.org/documentation/configuration/wireless/access-point-routed.md
+# Enable IPv4 routing
+net.ipv4.ip_forward=1
+```
+Enabling routing will allow hosts from network `192.168.4.0/24` to reach the LAN and the main router towards the internet. In order to allow traffic between clients on this foreign wireless network and the internet without changing the configuration of the main router, the Raspberry Pi can substitute the IP address of wireless clients with its own IP address on the LAN using a "masquerade" firewall rule.
```

Comment:
@JamesH65 Neither available copy-editor understands "foreign" here :) Please can you confirm that it's correct?

Reply:
That sounds like it should be on its own page - I would expect it to be called something like "Advanced - routing and IP masquerading. Normally wireless access points just bridge onto the local ethernet network.

Reply:
That needs reworded. "Foreign" here assumes that "native" is the wired LAN segment, and the user requires routing to be set up in order for things to work, since they have some setup that won't route traffic unless it comes from a specific IP address range.

Reply:
I cannot comment one way or the other, as this is out of my level of knowledge.

Reply:
@epoch1970 Have you got time to clarify this?

Reply:
Routers route or drop traffic whether they have a relevant route for it, or not. Firewalls filter traffic that isn't dropped.
Between a source and a destination host (and back again), each router along the path makes a _fatal_ decision (pass/drop) based on the _local information_ it has.

Say a wifi client 192.168.4.99 queries Google's DNS 8.8.8.8, and receives a response; The path includes 2 local routers:  Pi > ISP router > (internet routers) > ISP router > Pi. Any router can break the chain.
Internet routers automatically exchange routes, they know their stuff. Local routers require local administration...
 - The Pi router knows all networks involved, it will let traffic pass. 
 - The ISP router does not know about network 192.168.4.0/24, it _will_ drop traffic. The simplest and most portable solution is to hide the hosts in this network under the identity of the Pi, using the well-known LAN address 10.10.0.2.

Tutorials all choose this double-NAT solution. None (I think) try to explain the reason for  masquerading. So let me try again:

Enabling routing will allow hosts from network `192.168.4.0/24` to reach the LAN and the main router towards the internet. 
In order to ensure that the main router will allow traffic between wireless clients and the internet, the Raspberry Pi can substitute their IP address with its own IP address on the LAN using a "masquerade" firewall rule.

Hope that's more clear.

Reply:
@epoch1970 I'm not sure that an explanation of masquerading / routing is what was being asked for?

To answer @helenlynn 's original question, I think "foreign wireless network" here is referring to the wireless network that's being managed exclusively by the RPi, rather than the wireless network being managed by the user's "main router". I wonder if "new wireless network" would be a suitable replacement? :man_shrugging:

Reply:
Given the question and previous remarks, I think the long reply might be on point.
As I said, in the previous tutorial, you can read "Add a masquerade for outbound traffic on eth0:". You can leave it at that if you prefer. Proven and native.

Reply:
I've made all other changes requested but am not sure what to do here, any suggestions?

Reply:
@lurch 's suggestion of "new wireless network" is possibly suitable if it is technically correct, which I suspect that it is, but I'm not in a position to be sure.

Reply:
No that suggestion was not correct. The fact the new network is wireless is irrelevant. What counts is that the new IP network (192.168.4.0) is different from the one managed by the main router (10.0.0.0). See my answer above for details.

Reply:
I think the solution is to include some discussion about what "routed" means in this context, at the start of the page. You then just need to call the subnet something like "the routed subnet" consistently throughout the page (rather than foreign).

Also, a minor nit: it is common practice when using /24 networks to put routers at the 'top' of the subnet, starting at .254 downwards, leaving the low-numbered addresses (.1, .2 etc) for the "normal" hosts.

Reply:
> it is common practice when using /24 networks to put routers at the 'top' of the subnet, starting at .254 downwards, leaving the low-numbered addresses (.1, .2 etc) for the "normal" hosts.

Oh? Most of the home routers I've seen default to something like 192.168.0.1 or 192.168.1.1

Reply:
> > it is common practice when using /24 networks to put routers at the 'top' of the subnet, starting at .254 downwards, leaving the low-numbered addresses (.1, .2 etc) for the "normal" hosts.
> 
> Oh? Most of the home routers I've seen default to something like 192.168.0.1 or 192.168.1.1

While it is certainly true that most home routers give themselves addresses ending in .1, that's not really how it's meant to be done. BT manage to get it right though. Like I said, it's a minor nit, which I am picking 😋